src/editor.js

```diff
@@ -622,6 +622,19 @@ function initEventHandlers() {
     // **********************************
     // **     Toolbar - right side     **
     // **********************************
+    // Project Name listing
+    // Make the text in the project-name span editable
+    $('.project-name').attr('contenteditable', 'true')  
+            // Change the styling to indicate to the user that they are editing this field        
+            .on('focus', () => {
+                $('.project-name').addClass('project-name-editable');
+            })
+            // reset the style and save the new project name to the projectData object 
+            .on('blur', () => {
+                $('.project-name').removeClass('project-name-editable');
```

Comment:
It would be good to call a validator function to ensure that the user input is clean and does not exceed the project name field width before we actually accept the edit.

Reply:
Easy enough to do - is there a particular ruleset?  We do filter/replace/truncate before generating a filename, and I'm not sure that there are any kind of limits imposed by an HTML field or a javascript variable, so any restrictions would be our call to make.  Just let me know what we should apply here and I'll code it in.

Reply:
My thought was that we should try our best to show the user the filename that will appear in their file system when they save the project. 

I think the length of the filename was limited by the database field size in BlocklyProp. Here, the real limit is based on the underlying filesystem. From the UI prospective, the project name field has to have some maximum width. The project name can be longer than that width but then we get into scrolling the name within the field. Good Times there. 

The allowed characters is also OS dependent. It looks like we are doing some really basic filtering in editor.js: downloadCode()

```
replace(/[^a-z0-9]/gi, '_').toLowerCase();
```

We should have a function that takes a proposed project/filename string and return one that is sanitized or null if it cannot be sanitized.

Reply:
No scrolling on that <div>!  If it's too long, we can truncate it's display and add "..." to the end.  How's 100 chars sound?  Arbitrary, but should be sufficient.  We'll display up to 30 before adding ellipses.
Just in case we need it, here's how to hide scrollbars but still allow scrolling:
```
.container {
    overflow-y: scroll;
    scrollbar-width: none; /* Firefox */
    -ms-overflow-style: none;  /* IE 10+ */
}
.container::-webkit-scrollbar { /* WebKit */
    width: 0;
    height: 0;
}
```

Reply:
I agree that there should not be any scroll bars in the project name UI. I was thinking that the text would just scroll off to the left as one enters a huge project name. Once the user accepts that change, the huge project name snaps back to the beginning. That's the normal behavior of a text field, right? Except we may not actually be using a text field - although that might be easier to implement.

I'm ok with capping the max characters to 100. Can you make that value a constant and then reference the constant? It will make our lives easier 6 months from now when we need to change it.

Reply:
Okay - char limits have been implemented in the latest commit.  I had to refactor the UI css/styling a bit to make this work properly.

I also created two constants, and capitalized some other constants in the file to to make them visually obvious that they are constants, and that makes it consistent with Blockly's code style guide as well.